src/backend/query_builder.rs

```diff
@@ -1261,53 +1272,8 @@ pub trait QueryBuilder: QuotedBuilder + EscapeBuilder + TableRefBuilder {
     #[doc(hidden)]
     /// Translate part of a condition to part of a "WHERE" clause.
     fn prepare_condition_where(&self, condition: &Condition, sql: &mut dyn SqlWriter) {
-        if condition.negate {
-            write!(sql, "NOT ").unwrap()
-        }
-
-        if condition.is_empty() {
-            match condition.condition_type {
-                ConditionType::All => self.prepare_constant_true(sql),
-                ConditionType::Any => self.prepare_constant_false(sql),
-            }
-            return;
-        }
-
-        if condition.negate {
-            write!(sql, "(").unwrap();
-        }
-        condition.conditions.iter().fold(true, |first, cond| {
-            if !first {
-                match condition.condition_type {
-                    ConditionType::Any => write!(sql, " OR ").unwrap(),
-                    ConditionType::All => write!(sql, " AND ").unwrap(),
-                }
-            }
-            match cond {
-                ConditionExpression::Condition(c) => {
-                    if condition.conditions.len() > 1 {
-                        write!(sql, "(").unwrap();
-                    }
-                    self.prepare_condition_where(c, sql);
-                    if condition.conditions.len() > 1 {
-                        write!(sql, ")").unwrap();
-                    }
-                }
-                ConditionExpression::SimpleExpr(e) => {
-                    if condition.conditions.len() > 1 && (e.is_logical() || e.is_between()) {
-                        write!(sql, "(").unwrap();
-                    }
-                    self.prepare_simple_expr(e, sql);
-                    if condition.conditions.len() > 1 && (e.is_logical() || e.is_between()) {
-                        write!(sql, ")").unwrap();
-                    }
-                }
-            }
-            false
-        });
-        if condition.negate {
-            write!(sql, ")").unwrap();
-        }
+        let simple_expr = condition.to_simple_expr();
+        self.prepare_simple_expr(&simple_expr, sql);
```

Comment:
This is very cool if we managed to not break any Condition based test cases!

Reply:
There are some tests that had to be fixed, but there was no way around it unfortunately. 
I have had a go with sea-orm - and it appears to have no tests broken at least.